### PR TITLE
Attempt to publish binary on release

### PR DESCRIPTION
**Pull request by bam241**
this is an attempt to publish out build on release.

not sure how that will work tho.... or how to test it....

### CODE REVIEW

**Comment by bam241** [comment]
this should be working now and produce something like: https://github.com/bam241/Trelis-plugin/releases/tag/0.0.8
(upload to come as CI finishes)

File: .github/workflows/unix.yml [hunk 1]
```diff
@@ -132,10 +134,18 @@ jobs:
...
+          path: ~/svalinn-plugin_${{ matrix.os }}_cubit_${{ matrix.cubit }}.tgz
+    
+      - if: github.event_name == 'release'
+        name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ~/svalinn-plugin_${{ matrix.os }}_cubit_${{ matrix.cubit }}.tgz
+          asset_name: svalinn-plugin_${{ matrix.os }}_cubit_${{ matrix.cubit }}.tgz
+          tag: ${{ github.ref }}
```
Comment (gonuke) [inline .github/workflows/unix.yml]: add a newline

File: .github/workflows/unix.yml [hunk 2]
```diff
@@ -132,10 +134,18 @@ jobs:
         run: |
           ${system}_build_plugin_pkg ${{ matrix.cubit }}
 
-      - name: Upload package
+      - if: github.event_name != 'release'
+        name: Upload package
```
Comment (gonuke) [inline .github/workflows/unix.yml:139]: where does this get uploaded?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @bam241 - looking forward to getting this working!

File: .github/workflows/unix.yml (hunk 2 above)
Comment (bam241) [reply .github/workflows/unix.yml:139]: should have been upload there: https://github.com/bam241/Trelis-plugin/releases/tag/0.0.8

but there is a problem....

Comment (bam241) [reply .github/workflows/unix.yml:139]: basically we should end up with something like:
https://github.com/bam241/Trelis-plugin/releases/tag/0.0.7

**Comment by bam241** [comment]
this should be ready,
you can see an example of use on my fork:
item get added automatically here: https://github.com/bam241/Trelis-plugin/releases/tag/0.0.12
when each job on CI get done they will be added in the release page
(corresponding jobs: https://github.com/bam241/Trelis-plugin/actions/runs/943758101)

**Comment by bam241** [comment]
cc @gonuke

**Comment by bam241** [comment]
the action I added automatically runs on new release, and add the artefact as binaries in the release, (where the release has been created)

I ran it on my fork to test it and avoid spamming and polluting `develop` here, but this is not kink to a fork, will add the artifacts where the release has been issued

File: .github/workflows/unix.yml (hunk 2 above)
Comment (pshriwise) [reply .github/workflows/unix.yml:139]: Wait I'm confused too, this block gets executed when the event is not a release right? Do we want to upload anything in that case?

Comment (bam241) [reply .github/workflows/unix.yml:139]: this is upload the artifact at the job summary location they're are useful for debug when running CI, 
on release we use the other upload mechanism

Comment (pshriwise) [reply .github/workflows/unix.yml:139]: Ahhh I see. Very nice! Didn't know that was a thing.

**Review by pshriwise — APPROVED** [review]
Great addition @bam241! So nice to have these automatically published in a place that's easily accessible for new users.

**Review by gonuke — APPROVED** [review]
LGTM - thanks @bam241

**Pull request merged**